[Qt] fix coincontrol sort issue

jonasschnelli opened the pull request:
Reported by @gmaxwell 
The sort order for the amount column is lexicographical resulting in a order like "1 10 2 20 200 3 ...".

The attempts of this PR is to fix this.

jonasschnelli committed:
[Qt] fix coincontrol sort issue

laanwj commented on src/qt/coincontroldialog.cpp:
```diff
@@ -721,11 +730,11 @@ void CoinControlDialog::updateView()
             // amount
             itemOutput->setText(COLUMN_AMOUNT, BitcoinUnits::format(nDisplayUnit, out.tx->vout[out.i].nValue));
-            itemOutput->setText(COLUMN_AMOUNT_INT64, strPad(QString::number(out.tx->vout[out.i].nValue), 15, " ")); // padding so that sorting works correctly
+            itemOutput->setData(COLUMN_AMOUNT_INT64, Qt::DisplayRole, QVariant((qlonglong)out.tx->vout[out.i].nValue)); // padding so that sorting works correctly
```
Let's get rid of the hidden _INT64 columns completely, they should no longer be necessary.

laanwj replied:
Also: function strPad can go after this.

laanwj commented:
Working on a patch

laanwj committed:
[Qt] Clean up and fix coincontrol tree widget handling

- Do sorting for date, amount and confirmations column as longlong, not
  unsigned longlong.
- Use `UserRole` to store our own data. This makes it treated as
  ancillary data prevents it from being displayed.
- Get rid of `getMappedColumn` `strPad` - these are no longer necessary.
- Get rid of hidden `_INT64` columns.
- Start enumeration from 0 (otherwise values are undefined).

laanwj commented:
Please cherry-pick the top commit from: https://github.com/laanwj/bitcoin/tree/2016_11_trol_cleanup into this. I've done some further cleanups and fixes:

- Do sorting for date, amount and confirmations column as longlong, not unsigned longlong.
- Use `UserRole` to store our own data for sorting. This makes it treated as  ancillary data prevents it from being displayed.
- Get rid of `getMappedColumn` `strPad` - these are no longer necessary.
- Get rid of hidden `_INT64` columns.
- Start enumeration from 0 (otherwise values are undefined).

jonasschnelli commented:
Thanks @laanwj!
Cherry picked your 4231032

laanwj commented:
@gmaxwell Can you please test this in your environment and confirm whether the problem is fixed?

jonasschnelli commented:
Tested ACK (Ubuntu) after @laanwj commit.
Binaries: https://bitcoin.jonasschnelli.ch/pulls/9185/

Waiting for @gmaxwell's re-test

<img width="735" alt="bildschirmfoto 2016-11-21 um 13 55 27" src="https://cloud.githubusercontent.com/assets/178464/20483425/403e6474-aff2-11e6-9a68-1e01908c8561.png">

gmaxwell commented:
tested ACK!
